clippy_lints/src/literal_string_with_formatting_args.rs

```diff
@@ -124,7 +128,11 @@ impl LateLintPass<'_> for LiteralStringWithFormattingArg {
                     pos.start += diff_len;
                     pos.end += diff_len;
 
-                    let start = fmt_str[..pos.start].rfind('{').unwrap_or(pos.start);
+                    let mut start = pos.start;
+                    while start < fmt_str.len() && !fmt_str.is_char_boundary(start) {
+                        start += 1;
+                    }
+                    let start = fmt_str[..start].rfind('{').unwrap_or(start);
```

Comment:
Is this the same as `fmt_str[..fmt_str.ceil_char_boundary(pos.start)].rfind('{')`?

Reply:
For what it's worth, it looks like the format parser offsets all spans by 1 (presumably because it's relative to the `"` quote), so that's what causes the index to be in the middle of a character sometimes. Just doing `pos.start - 1` instead of `pos.start` on the old deleted line (to undo the parser's offset) should fix this too as far as I can tell, but this over-conservative approach should be fine too I guess

Reply:
I'd prefer to keep it over-conservative if you don't mind. :wink: